R/calc_genpop.R

```diff
@@ -0,0 +1,96 @@
+#' Calculate General Population PEPFAR results
+#'
+#' Standardizes key populations groupings and calculates estimated general population result
+#' by subtracting key populations results from Total Numerator
+#'
+#' @param df PEPFAR MER structured dataset in data.frame format for calculation, such as the object created by ICPIutilities::read_msd()
+#' @importFrom dplyr %>%
+#'
+#' @return a simplified summarised dataframe with a standardized kpgroup variable and results for general population
+#' @export
+#'
+
+#'
+calc_genpop <- function(df) {
+  ## Function only takes a data.frame
+  # require(usethis)
+  # require(devtools)
+  # library(ICPIutilities)
+  # library(tidyverse)
+  # library(roxygen2)
+
+  if(is.data.frame(df)) {
+
+    ## code is written to take MSD columns as named in native dataset.  read_msd(df,to_lower=F) only
+    if(any(grepl("[[:upper:]]",names(df)))) {
```

Comment:
suggest cleaning this up. Rather than having `if` statements that run the whole function, stop the function if the inverse is TRUE. So at the top you would just have the following lines:

```
#checks
# 1. did user provide a data frame
# 2. code is written to take MSD columns as named in native dataset.  read_msd(df,to_lower=F) only

if(!is.data.frame(df))
  stop("Not a data frame")
if(any(grepl("[[:upper:]]",names(df))))
  stop("You may have converted variable names to lower case.  Try ICPIUtilities::read_msd(df,to_lower=FALSE)")

Reply:
done